Understanding the Vector Database and how the embeddings or numerical vectors are genrerated

In [ ]:
import requests
import numpy as np

def generate_embeddings(text):
    url = "https://api.euron.one/api/v1/euri/embeddings"
    headers = {
        "Content-Type": "application/json",
        "Authorization": "Bearer euri-37b9fff3358867bfbeab3928b39f637bb00e12d253459000b0794bd0c4177b7f"
    }
    payload = {
        "input": text,
        "model": "text-embedding-3-small"
    }

    response = requests.post(url, headers=headers, json=payload)
    data = response.json()
    
    embedding = np.array(data['data'][0]['embedding'])
    
    return embedding

text = "My name is Anshul"

embedding = generate_embeddings(text)

In [2]:
embedding

array([-0.01054326, -0.06418464, -0.03602973, ..., -0.01287962,
        0.00856862, -0.00359349])

Now, we try by using a source code for a model open source model taken from Hugging face

In [3]:
len(embedding)

1536

The above length shows that whatever text data we are going to pass through the selected model in the baove script will return the associated vector of dimension 1536 or length 1536. In 1536 dimension plance. In 1536 dimension plane, we can represent the complete text via that data at a single point

In [11]:
pip install sentence-transformers

  Using cached sentence_transformers-5.1.0-py3-none-any.whl.metadata (16 kB)
  Using cached transformers-4.55.4-py3-none-any.whl.metadata (41 kB)
  Using cached huggingface_hub-0.34.4-py3-none-any.whl.metadata (14 kB)
  Using cached hf_xet-1.1.8-cp37-abi3-macosx_11_0_arm64.whl.metadata (703 bytes)
  Using cached sympy-1.14.0-py3-none-any.whl.metadata (12 kB)
  Using cached tokenizers-0.21.4-cp39-abi3-macosx_11_0_arm64.whl.metadata (6.7 kB)
  Using cached safetensors-0.6.2-cp38-abi3-macosx_11_0_arm64.whl.metadata (4.1 kB)
Using cached sentence_transformers-5.1.0-py3-none-any.whl (483 kB)
Using cached huggingface_hub-0.34.4-py3-none-any.whl (561 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 73.6/73.6 MB 5.0 MB/s eta 0:00:0000:0100:01
Using cached transformers-4.55.4-py3-none-any.whl (11.3 MB)
Using cached hf_xet-1.1.8-cp37-abi3-macosx_11_0_arm64.whl (2.6 MB)
Using cached safetensors-0.6.2-cp38-abi3-macosx_11_0_arm64.whl (432 kB)
Using cached sympy-1.14.0-py3-none-any.whl (6.3 MB)
Using

In [14]:
# Requires transformers>=4.51.0
# Requires sentence-transformers>=2.7.0

from sentence_transformers import SentenceTransformer

# Load the model
model = SentenceTransformer("Qwen/Qwen3-Embedding-0.6B")

# The documents to embed
documents = [
    "The capital of China is Beijing.",
    "Gravity is a force that attracts two bodies towards each other. It gives weight to physical objects and is responsible for the movement of planets around the sun.",
]


document_embeddings = model.encode(documents)



modules.json:   0%|          | 0.00/349 [00:00<?, ?B/s]

config_sentence_transformers.json:   0%|          | 0.00/215 [00:00<?, ?B/s]

README.md: 0.00B [00:00, ?B/s]

config.json:   0%|          | 0.00/727 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/1.19G [00:00<?, ?B/s]

tokenizer_config.json: 0.00B [00:00, ?B/s]

vocab.json: 0.00B [00:00, ?B/s]

merges.txt: 0.00B [00:00, ?B/s]

tokenizer.json:   0%|          | 0.00/11.4M [00:00<?, ?B/s]

config.json:   0%|          | 0.00/313 [00:00<?, ?B/s]

In [15]:
document_embeddings[0]

array([-0.04718248, -0.0209168 ,  0.00366512, ...,  0.05624256,
        0.07076067, -0.01714627], dtype=float32)

In [16]:
len(document_embeddings[0])

1024

In [17]:
len(document_embeddings[1])

1024

In [ ]:
from sentence_transformers import SentenceTransformer

# Load the model
model = SentenceTransformer("Qwen/Qwen3-Embedding-0.6B")

# The documents to embed
documents = [
    "my name is Anshul and I am a wirelesss researcher",
    "Anshul is my name and I am a wireless researcher",
    "apple is red and it is a fruit",
    "banana is yellow and it is a fruit",
    "Anshul is taking a class of GenAI and going to attend the genai interview prepartion"
]


document_embeddings = model.encode(documents)

In [19]:
document_embeddings

array([[ 0.01977447, -0.04518175, -0.01356521, ..., -0.02208575,
         0.02351786,  0.00993915],
       [ 0.03461524, -0.06594328, -0.01397689, ..., -0.02677975,
         0.01179108,  0.01687081],
       [-0.08274745,  0.05246599, -0.00250338, ..., -0.00065248,
         0.01755737,  0.01571283],
       [-0.05386708,  0.01229258, -0.00333655, ..., -0.00775521,
        -0.03061855,  0.02123336],
       [ 0.01731795, -0.04901413, -0.01053595, ...,  0.02554622,
         0.01966246,  0.00361277]], dtype=float32)

In [20]:
# Compute the (cosine) similarity for the document embeddings
from numpy import dot
from numpy.linalg import norm


In [23]:
sent1 = document_embeddings[0]
sent2 = document_embeddings[1]
sent3 = document_embeddings[2]
sent4 = document_embeddings[3]
sent5 = document_embeddings[4]

In [28]:
dot(sent1, sent4)/(norm(sent1)*norm(sent4))

0.34250087

In [26]:
dot(sent1, sent3)/(norm(sent1)*norm(sent3))

0.34792316

In [ ]:
dot(sent1, sent2)/(norm(sent1)*norm(sent2))

In [29]:
query = ["whi is Anshul?"]

In [30]:
from sentence_transformers import SentenceTransformer

# Load the model
model = SentenceTransformer("Qwen/Qwen3-Embedding-0.6B")

# The documents to embed
query_embeddings = model.encode(query)

In [31]:
query_embeddings

array([[ 0.04078789, -0.08957411, -0.00965021, ...,  0.0226004 ,
        -0.0009318 ,  0.01917204]], dtype=float32)

In [32]:
dist1 = dot(query_embeddings, sent1)/(norm(query_embeddings)*norm(sent1))
dist2 = dot(query_embeddings, sent2)/(norm(query_embeddings)*norm(sent2))
dist3 = dot(query_embeddings, sent3)/(norm(query_embeddings)*norm(sent3))
dist4 = dot(query_embeddings, sent4)/(norm(query_embeddings)*norm(sent4))
dist5 = dot(query_embeddings, sent5)/(norm(query_embeddings)*norm(sent5))

In [33]:
dist1, dist2, dist3, dist4, dist5

(array([0.63197184], dtype=float32),
 array([0.6907879], dtype=float32),
 array([0.31062204], dtype=float32),
 array([0.30189916], dtype=float32),
 array([0.60016066], dtype=float32))